Rubberduck.Core/AutoComplete/AutoCompleteBlockBase.cs

```diff
@@ -73,14 +80,16 @@ public override bool Execute(AutoCompleteEventArgs e, AutoCompleteSettings setti
 
         public override bool IsMatch(string code)
         {
+            code = code.Trim().StripStringLiterals();
             var pattern = SkipPreCompilerDirective
                             ? $"\\b{InputToken}\\b"
                             : $"{InputToken}\\b"; // word boundary marker (\b) would prevent matching the # character
             var regexOk = MatchInputTokenAtEndOfLineOnly
-                ? code.EndsWith(InputToken, System.StringComparison.OrdinalIgnoreCase)
-                : Regex.IsMatch(code.Trim(), pattern, RegexOptions.IgnoreCase);
+                ? !code.StartsWith(Tokens.Else, System.StringComparison.OrdinalIgnoreCase) 
+                    && code.EndsWith(InputToken, System.StringComparison.OrdinalIgnoreCase)
+                : Regex.IsMatch(code, pattern, RegexOptions.IgnoreCase);
 
-            return regexOk && !code.HasComment(out _);
+            return regexOk && (code.HasComment(out int commentIndex) ? code.IndexOf(InputToken) < commentIndex : true);
```

Comment:
this is getting complicated. let's not use ternary operators here, okay?

Reply:
This also applies for the preceding code

Reply:
TBH I find the whole thing is becoming a mess, rather quickly. Something is amiss.